pkg/providers/gcp/provider_redis.go

```diff
@@ -244,34 +281,53 @@ func (p *RedisProvider) getRedisConfig(ctx context.Context, r *v1alpha1.Redis) (
 	defaultRegion, err := GetRegionFromStrategyOrDefault(ctx, p.Client, strategyConfig)
 	if err != nil {
 		errMsg := "failed to get default gcp region"
-		return nil, nil, nil, errorUtil.Wrap(err, errMsg)
+		return nil, errorUtil.Wrap(err, errMsg)
 	}
 	if strategyConfig.Region == "" {
 		p.Logger.Debugf("region not set in deployment strategy configuration, using default region %s", defaultRegion)
 		strategyConfig.Region = defaultRegion
 	}
-	createInstanceRequest, deleteInstanceRequest, err := p.buildRedisConfig(r, strategyConfig)
-	if err != nil {
-		return nil, nil, nil, errorUtil.Wrap(err, "failed to build redis config")
-	}
-	return createInstanceRequest, deleteInstanceRequest, strategyConfig, nil
+	return strategyConfig, nil
 }
 
-func (p *RedisProvider) buildRedisConfig(r *v1alpha1.Redis, strategyConfig *StrategyConfig) (*redispb.CreateInstanceRequest, *redispb.DeleteInstanceRequest, error) {
-	// TODO (MGDAPI-4667): create a request with all required fields
+func (p *RedisProvider) buildCreateInstanceRequest(ctx context.Context, r *v1alpha1.Redis, strategyConfig *StrategyConfig, networkAddress string) (*redispb.CreateInstanceRequest, error) {
 	createInstanceRequest := &redispb.CreateInstanceRequest{}
+	if err := json.Unmarshal(strategyConfig.CreateStrategy, createInstanceRequest); err != nil {
+		return nil, errorUtil.Wrap(err, "failed to unmarshal gcp redis create strategy")
+	}
+	if createInstanceRequest.Parent == "" {
+		createInstanceRequest.Parent = fmt.Sprintf(redisParentFormat, strategyConfig.ProjectID, strategyConfig.Region)
+	}
+	if createInstanceRequest.InstanceId == "" {
+		instanceID, err := resources.BuildInfraNameFromObject(ctx, p.Client, r.ObjectMeta, defaultGcpIdentifierLength)
+		if err != nil {
+			return nil, errorUtil.Wrap(err, "failed to build redis instance id from object")
+		}
+		createInstanceRequest.InstanceId = instanceID
+	}
+	if createInstanceRequest.Instance == nil {
+		createInstanceRequest.Instance = &redispb.Instance{
+			Name:              fmt.Sprintf(redisInstanceNameFormat, strategyConfig.ProjectID, strategyConfig.Region, createInstanceRequest.InstanceId),
+			Tier:              redispb.Instance_STANDARD_HA,
+			MemorySizeGb:      redisMemorySizeGB,
+			AuthorizedNetwork: strings.Split(networkAddress, "v1/")[1],
+			RedisVersion:      redisVersion,
+		}
+	}
```

Comment:
I think we should check for the presence of these and set them separately, similar to what is done in Postgres/AWS. This allows the user to provide a create strategy with fewer fields (`"createStrategy": { "tier": "BASIC" }`) and still retain CRO defaults for other values.

Reply:
I kind of expected we would have to set `ReservedIpRange` to the IP address range from the network reconciler - If there are multiple IP ranges is ours always chosen or is there a chance it picks the other one? If this does need to be set is the `AuthorizedNetwork` field required or redundant?